gcp_variant_transforms/transforms/infer_headers_test.py

```diff
@@ -401,3 +417,141 @@ def test_pipeline(self):
       assert_that(inferred_headers,
                   asserts.header_fields_equal_ignore_order([expected]))
       p.run()
+
+  def test_infer_annotation_types_no_conflicts(self):
+    anno_fields = ['CSQ']
+    header = self._get_sample_header_fields_with_annotation()
+    variant = self._get_sample_variant_1()
+    variant.info['CSQ'] = ['A|GENE1|100|1.2', 'TT|GENE1|101|1.3']
+    infer_header_fields = infer_headers._InferHeaderFields(False, anno_fields)
+    new_header = next(infer_header_fields.process(variant, header))
+    desc = 'Inferred type field for annotation {}.'
+    expected = vcf_header_io.VcfHeader(infos={
+        'CSQ_Gene_TYPE':
```

Comment:
I think some of the checks here are unnecessary (e.g. if we change the description format, we have to change all of these tests too; can be saved by reusing a constant, but I think it's not needed).
What do you think of the following approach:

Create a dict like:
```
expected_types = {
    'CSQ_Gene_TYPE': 'String',
    'CSQ_Position_TYPE': 'Integer', ...
}
```
and then just loop through the inferred_headers and check that the types match. You can also check the length of the two dicts to make sure they match.